lib/Driver/Driver.cpp

```diff
@@ -1636,6 +1649,11 @@ bool Driver::handleImmediateArgs(const ArgList &Args, const ToolChain &TC) {
     printVersion(TC, llvm::errs());
     SuppressNoInputFilesError = true;
   }
+  
+  /* SR-5860: "-e" doesn't expect any Input Files */
+  if (Args.hasArg(options::OPT_e)) {
+    SuppressNoInputFilesError = true;
```

Comment:
We probably need to do the opposite here: emit a diagnostic if there are both {{-e}} options and input files. (This can happen at the same place as the check for multiple input files.)

The actual diagnostics are defined in DiagnosticsDriver.def.